src/ProxyManager/ProxyGenerator/PropertyGenerator/PublicPropertiesMap.php

```diff
@@ -38,12 +38,13 @@ class PublicPropertiesMap extends PropertyGenerator
 
     /**
      * @param \ReflectionClass $originalClass
+     * @param array            $excludedProperties
      */
-    public function __construct(ReflectionClass $originalClass)
+    public function __construct(ReflectionClass $originalClass, array $excludedProperties = [])
```

Comment:
Can we re-design this to accept a `Properties` instance instead of `$originalClass`?